### ISSUE TITLE
DelayProcedure which implements InputProcedure

### ISSUE THREAD

**Issue by golanb** [ProcedureKit/ProcedureKit#890]
Hi,

The scenario is this:
- Tell the server to start converting a file (NetworkDataProcedure). The server responds with the estimated time for the conversion to complete
- Wait for the conversion to complete (DelayProcedure)
- Download the converted file (NetworkDownloadProcedure).

Is there a way to use DelayProcedure and inject the delay interval as an input instead of passing it on init?

Thanks!

**Comment by danthorpe** [comment]
Hi @golanb,

Well, this sounds like a scenario for a `GroupProcedure`. In pseudo code, you effectively want to do:

1. create a network request to perform the download
2. decode the response to extract the estimated time
3. inject the estimated time (with some tolerance?) into a `DelayProcedure`
4. inject the file destination into a `NetworkDownloadProcedure`
5. add the delay as a dependency of the download procedure.

All of this you can _package_ up into a `GroupProcedure` subclass, so that you init (or inject) the URL, and output the downloaded file (or its location on disk).

Do you have a sample of the 1st network response?

**Comment by danthorpe** [comment]
But yeah, if `DelayProcedure` conformed to `InputProcedure` it would certainly make the scenario easier.

**Comment by golanb** [comment]
Hi @danthorpe ,

That's exactly what I was trying to do. The only limitation is that I cannot inject the estimated time value from the first network procedure into the delay procedure because delay procedure does not conform to InputProcedure. Here is some sample code to illustrate the problem:

`class DownloadConvertedFileProcedure: GroupProcedure {
    
    init(fileId: String) {
        let conversionStartProcedure = ConversionStartProcedure(fileId)
        let conversionWaitProcedure = DelayProcedure()
        let downloadFileProcedure = DownloadFileProcedure(fileId)
        
        // The next line is what I wish I could do with a DelayProcedure
        conversionWaitProcedure.injectResult(from: conversionStartProcedure) { status in
            return status.estimatedTime
        }
        
        downloadFileProcedure.addDependency(conversionWaitProcedure)
        
        let operations = [
            conversionStartProcedure,
            conversionWaitProcedure,
            downloadFileProcedure
        ]
        super.init(operations: operations)
    }
}`

**Comment by danthorpe** [comment]
@golanb hey, #891 is a PR which adds support to `DelayProcedure` - you could checkout your project against it. You could then do this:

```swift
// The next line is now possible
conversionWaitProcedure.injectDelay(from: conversionStartProcedure) { $0.estimatedTime }
```

This will get merged into `development` branch this week - and will be released either in 5.0.0 _final_ build, or 5.1.x.

**Comment by golanb** [comment]
Super! Thanks!
